components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -1,19 +1,340 @@
+func (src *Subscription) ConvertTo(dstRaw conversion.Hub) error { // nolint
+	dst, ok := dstRaw.(*v1alpha2.Subscription)
+	if !ok {
+		return errors.Errorf(hubVersionErrMsg)
+	}
+	return v1ToV2(src, dst)
+}
+
+// v1ToV2 copies the v1alpha1-type field values into v1alpha2-type field values
+func v1ToV2(src *Subscription, dst *v1alpha2.Subscription) error {
+
+	// ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	// SPEC fields
+
+	dst.Spec.ID = src.Spec.ID
+	dst.Spec.Sink = src.Spec.Sink
+	dst.Spec.Source = ""
+
+	src.setV2TypeMatching(dst)
+
+	// protocol fields
+	src.setV2ProtocolFields(dst)
+
+	// Types
+	if err := src.setV2SpecTypes(dst); err != nil {
+		return err
+	}
+
+	// Config
+	src.natsSpecConfigToV2(dst)
+
+	// STATUS Fields
+
+	// Ready
+	dst.Status.Ready = src.Status.Ready
+
+	// Conditions
+	for _, condition := range src.Status.Conditions {
+		dst.Status.Conditions = append(dst.Status.Conditions, ConditionV1ToV2(condition))
+	}
+
+	// event types
+	src.setV2StatusTypes(dst)
+
+	// Backend-specific status
+
+	// BEB
+	src.bebBackendStatusToV2(dst)
+
+	// NATS
+	src.natsBackendStatusToV2(dst)
+
+	return nil
+}
+
+// ConvertFrom converts this Subscription from the Hub version (v2).
+func (dst *Subscription) ConvertFrom(srcRaw conversion.Hub) error { // nolint
+	src, ok := srcRaw.(*v1alpha2.Subscription)
+	if !ok {
+		return errors.Errorf(hubVersionErrMsg)
+	}
+	return v2ToV1(dst, src)
+}
+
+// v2ToV1 copies the v1alpha2-type field values into v1alpha1-type field values
+func v2ToV1(dst *Subscription, src *v1alpha2.Subscription) error {
+
+	// ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Spec.ID = src.Spec.ID
+	dst.Spec.Sink = src.Spec.Sink
+
+	dst.setV1ProtocolFields(src)
+
+	dst.Spec.Filter = &BEBFilters{
+		Filters: []*BEBFilter{},
+	}
+
+	for _, eventType := range src.Spec.Types {
+		filter := &BEBFilter{
+			EventSource: &Filter{
+				Property: "source",
+				Type:     fmt.Sprint(v1alpha2.EXACT),
+				Value:    src.Spec.Source,
+			},
+			EventType: &Filter{
+				Type:     fmt.Sprint(v1alpha2.EXACT),
+				Property: "type",
+				Value:    eventType,
+			},
+		}
+		dst.Spec.Filter.Filters = append(dst.Spec.Filter.Filters, filter)
+	}
+
+	if src.Spec.Config != nil {
+		dst.natsSpecConfigToV1(src)
+	}
+
+	// Conditions
+	for _, condition := range src.Status.Conditions {
+		dst.Status.Conditions = append(dst.Status.Conditions, ConditionV2ToV1(condition))
+	}
+
+	dst.Status.Ready = src.Status.Ready
+
+	dst.setV1CleanEvenTypes(src)
+	dst.bebBackendStatusToV1(src)
+	dst.natsBackendStatusToV1(src)
 
 	return nil
 }
 
-func (s *Subscription) ConvertFrom(rawObj conversion.Hub) error {
-	_ = rawObj.(*v1alpha2.Subscription)
+// setV2TypeMatching sets the default typeMatching on the v1alpha2 Subscription version
+func (src *Subscription) setV2TypeMatching(dst *v1alpha2.Subscription) {
+	dst.Spec.TypeMatching = v1alpha2.EXACT
+}
+
+// setV2ProtocolFields converts the protocol-related fields from v1alpha1 to v1alpha2 Subscription version
```

Comment:
nit: Please use the godot linter (from golangci_lint) to find the other missing dots :)
```suggestion
// setV2ProtocolFields converts the protocol-related fields from v1alpha1 to v1alpha2 Subscription version.
```